daft/execution/execution_step.py

```diff
@@ -947,6 +947,40 @@ def run_partial_metadata(self, input_metadatas: list[PartialPartitionMetadata])
         ]
 
 
+@dataclass(frozen=True)
+class CrossJoin(SingleOutputInstruction):
+    left_in_outer_loop: bool
+
+    def run(self, inputs: list[MicroPartition]) -> list[MicroPartition]:
+        return self._cross_join(inputs)
+
+    def _cross_join(self, inputs: list[MicroPartition]) -> list[MicroPartition]:
+        # All inputs except for the last are the left side of the join, in order to support left-broadcasted joins.
```

Comment:
If this is true why is it 

`        left, right = inputs`?

won't `left` and `right` just be `inputs[0]` and `inputs[1]`

when `left` should be`inputs[..-1]`, and right should be `inputs[-1]`

(not sure using right Python syntax)

Reply:
oh I forgot to remove this comment when I copied it over. good catch